cql3/statements/alter_keyspace_statement.cc

```diff
@@ -45,15 +45,17 @@ future<> cql3::statements::alter_keyspace_statement::check_access(query_processo
     return state.has_keyspace_access(_name, auth::permission::ALTER);
 }
 
-static unsigned get_abs_rf_diff(const std::string& curr_rf, const std::string& new_rf) {
+static int parse_rf(const std::string& rf) {
     try {
-        return std::abs(std::stoi(curr_rf) - std::stoi(new_rf));
-    } catch (std::invalid_argument const& ex) {
-        on_internal_error(mylogger, fmt::format("get_abs_rf_diff expects integer arguments, "
-                                                "but got curr_rf:{} and new_rf:{}", curr_rf, new_rf));
-    } catch (std::out_of_range const& ex) {
-        on_internal_error(mylogger, fmt::format("get_abs_rf_diff expects integer arguments to fit into `int` type, "
-                                                "but got curr_rf:{} and new_rf:{}", curr_rf, new_rf));
+        const auto value = std::stoi(rf);
+        if (value < 0) {
+            on_internal_error(mylogger, std::format("parse_rf() expects a valid RF value, but it got: {}", rf));
```

Comment:
parse_rf() is called on user input, so it's not appropriate to raise internal error. I think it should be a regular exception.